e2e_tests/tests/cluster/test_resource_manager.py

```diff
@@ -0,0 +1,109 @@
+import os
+import shutil
+import tempfile
+import time
+from typing import List
+
+import pytest
+import yaml
+
+from determined.common.api import bindings
+from determined.common.api.bindings import determinedexperimentv1State
+from tests import config as conf
+from tests import experiment as exp
+
+from ..experiment.experiment import determined_test_session, num_active_trials
+from .test_agent_disable import _wait_for_slots
+
+# How long we should for the Nth = 1 rank to free.
+RANK_ONE_WAIT_TIME = 300
+
+
+@pytest.mark.e2e_cpu_2a
+@pytest.mark.timeout(300)
```

Comment:
maybe make this timeout be a bit larger than `RANK_ONE_WAIT_TIME`? then if it times out, it'd do it on `pytest.fail` which will print a nicer error